### DISCUSSION TITLE
Help on rule for getting 5 sensor values into a variable

### DISCUSSION THREAD

**Discussion by MacSass** [arendst/Tasmota#14355] [Q&A] [answered]
Hello,
I´m asking for help with building a rule which would put the values of the 5 sensors connected to my device into a variable.

Best would be to get the complete output of "status 8" command into a variable, as I want to push that full string to my webserver with websend (can´t use MQTT).
The output of "status 8" is:
> STATUS8 = {"StatusSNS":{"Time":"2022-01-10T00:21:39","Switch1":"OFF","SI7021":{"Temperature":21.6,"Humidity":53.3,"DewPoint":11.7},"BMP280":{"Temperature":22.4,"Pressure":1005.5,"SeaPressure":1011.6},"PressureUnit":"hPa","TempUnit":"C"}}
> 

I can get the value of a single sensor into a variable like this:
> Rule1 ON tele-SI7021#Temperature DO var1 %value% ENDON
> 
But then %var1% only holds the single temperature sensor value.
Instead of making 5 rules, I thought I could get the output of "status 8" by doing a rule like this (documentation states I can use any JSON output as trigger?):

> Rule2 ON StatusSNS DO var1 %value% ENDON
>
But this does not work - the rule is not triggered. Also this rule does not trigger when telemetry happens:

> Rule2 ON tele-StatusSNS DO var1 %value% ENDON
> 

How can I get the values of my sensors into a variable, so I can use it to push it to my webserver with websend regulary?
Either capturing output of telemetry "Sensor" or by triggering "status 8" every couple of minutes?

Or is there any other way to get the values of the sensors in a rule that I have missed?
Thanks for your help!

**Comment by barbudor** [comment, marked as answer]
Unfortunately with rules there is no other way than picking up each sensor value one by one in a different `var` and thenon the latest perform the WebSend/WebQuery with a concatenation
It might be possible with Scripting-Language but I don't know it enough.

If you are using an ESP32, you could do it quite easily in Berry language

**Reply by MacSass** [reply]
Thanks for the feedback - maybe this is something for future improvement.
I´ll take a look at "Berry language" as well, because I had not heard about that before. ...

**Reply by barbudor** [reply]
Berry is only on ESP32 (and variants incuding ESP32C3, ESP32S2)

Scripting-Language is an alternative to rules (both cannot co-exist) but is supported an all processors

**Reply by MacSass** [reply]
Took a quick look at Berry language - was not aware of it. Looks interesting, eventually I will switch over to ESP32 then ... don´t want to go for scripting on ESP8266 as I want to stick with pre-compiled builds.
Thanks for the inputs
